src/components/CheckBox/CheckBox.tsx

```diff
@@ -0,0 +1,56 @@
+import classNames from "classnames";
+import React from 'react';
+import CheckIcon from 'components/icons/CheckIcon';
+
+import styles from './CheckBox.module.scss'
+
+export type CheckBoxProps = Omit<
+  React.InputHTMLAttributes<HTMLInputElement>,
+  'onChange'
+> & {
+  /** Вызывается при клике на чекбокс */
+  onChange: (checked: boolean) => void;
+};
+
+const CheckBox: React.FC<CheckBoxProps> = (props) => {
+  const { onChange, className, checked, disabled } = props;
+
+  const [isChecked, setIsChecked] = React.useState<boolean>(checked ?? false);
```

Comment:
вижу в нескольких местах повторяется код `checked ?? false` - можно сразу при деструктуризации присвоить значение опциональному пропсу:
```suggestion
  const { onChange, className, checked = false, disabled } = props;

  const [isChecked, setIsChecked] = React.useState<boolean>(checked);
```